src/Engine/Waypoint/Waypoints.cpp

```diff
@@ -114,6 +119,11 @@ Waypoints::WaypointNameTree::Remove(const WaypointPtr &wp)
   TCHAR normalized_name[wp->name.length() + 1];
   NormalizeSearchString(normalized_name, wp->name.c_str());
   RadixTree<WaypointPtr>::Remove(normalized_name, wp);
+  if (!wp->shortname.empty()) {
+    TCHAR *normalized_shortname = new TCHAR;
+    NormalizeSearchString(normalized_shortname, wp->shortname.c_str());
```

Comment:
This will also crash, because `normalized_shortname` has room for exactly one character, i.e. barely enough for the null terminator.